data/Store.ts

```diff
@@ -705,7 +705,7 @@ export class Store extends HoistBase {
     /** True if the store has changes which need to be committed. */
     @computed
     get isDirty(): boolean {
-        return this._current !== this._committed;
+        return !equal(this._current, this._committed);
     }
```

Comment:
So there is a whole "normalization" step (see RecordSet.normalize)  that is supposed to do this evaluation after every update to `_current` and `_committed` and keep them pointer equivalent.    the fact that the getter is computed, might mean that this is an efficient way to have the same outcome, although I will say that one benefit of the normalization step is that it just frees up memory after an edit gets committed.